From 84d5a847796d7d7e6215e668cd08f312cdf9b10a Mon Sep 17 00:00:00 2001
From: "Philip, Avinash" <avinashphilip@ti.com>
Date: Tue, 13 Mar 2012 11:12:19 +0530
Subject: [PATCH 0906/1092] ARM: OMAP2+: am33xx: Setup eCAP pin-mux on
 initialization

Pinmuxing done from late_initcall will fail if CONFIG_DEBUG_FS is not
enabled. Hence move pin_mux setup to profile_detection stage.

Signed-off-by: Philip, Avinash <avinashphilip@ti.com>
---
 arch/arm/mach-omap2/board-am335xevm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-omap2/board-am335xevm.c b/arch/arm/mach-omap2/board-am335xevm.c
index ba7604d..ced0cc9 100644
--- a/arch/arm/mach-omap2/board-am335xevm.c
+++ b/arch/arm/mach-omap2/board-am335xevm.c
@@ -900,6 +900,7 @@ static struct pinmux_config wl12xx_pin_mux_evm_rev1_0[] = {
 static void enable_ecap0(int evm_id, int profile)
 {
 	backlight_enable = true;
+	setup_pin_mux(ecap0_pin_mux);
 }
 
 /* Setup pwm-backlight */
@@ -916,7 +917,6 @@ static int __init ecap0_init(void)
 	int status = 0;
 
 	if (backlight_enable) {
-		setup_pin_mux(ecap0_pin_mux);
 		platform_device_register(&am335x_backlight);
 	}
 	return status;
-- 
1.7.11.2

